packages/components/src/Chips/InternalChip.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from "react";
+import classnames from "classnames";
+import styles from "./InternalChip.css";
+import { useAssert } from "./useAssert";
+import { InternalChipProps } from "./ChipTypes";
+import { InternalChipAffix } from "./InternalChipAffix";
+import { Typography } from "../Typography";
+import { Tooltip } from "../Tooltip";
+
+export function InternalChip({
+  label,
+  active = false,
+  disabled = false,
+  invalid = false,
+  prefix,
+  suffix,
+  warnOnLongLabels = false,
+  onClick,
+}: InternalChipProps) {
+  const [truncateRef, setTruncateRef] = useState<HTMLElement | null>();
+  const isClickable = onClick && !disabled;
+  const classNames = classnames(styles.chip, {
+    [styles.clickable]: isClickable,
+    [styles.active]: active,
+    [styles.disabled]: disabled,
+    [styles.invalid]: invalid,
+    [styles.hasPrefix]: prefix,
+    [styles.hasSuffix]: suffix,
+  });
+  const affixProps = {
+    active: active,
+    invalid: invalid,
+    disabled: disabled,
+  };
+  const Tag = onClick ? "button" : "div";
+
+  const chip = (
+    <Tag
+      className={classNames}
+      {...(isClickable && {
+        onClick: onClick,
+        disabled: disabled,
+      })}
+      data-testid="chip-wrapper"
+    >
+      <InternalChipAffix affix={prefix} {...affixProps} />
+      <Typography numberOfLines={1} size="base">
+        <span ref={setTruncateRef}>{label}</span>
+      </Typography>
+      <InternalChipAffix affix={suffix} {...affixProps} />
+    </Tag>
+  );
+
+  useAssert(
+    warnOnLongLabels && label.length > 24,
+    `"${label}" label is too long for a Chip; you might be better off using Checkbox or Radio. ${label.length}/24 characters.`,
+    { warn: true },
+  );
+
+  return isTruncated() ? <Tooltip message={label}>{chip}</Tooltip> : chip;
```

Comment:
Instead of defining `chip` I think it makes more sense to do something like this:

```tsx
return isTruncated() ? (
  <Tooltip />
) : (
  <Tag>...</Tag>
)

```

Reply:
hmm.. but the tooltip has to wrap the chip content for it to show properly. no?